test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,344 @@
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"files\": [");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"jquery.js\",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"jquery.min.js\"");
+
+            _libManConfig.Save();
+            string cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib", "jquery");
+            LibmanTestsUtility.WaitForRestoredFiles(cwd, expectedFilesAndFolders, caseInsensitive: true);
+
+            Editor.Edit.DeleteToBeginningOfLine();
+            _libManConfig.Save();
+            LibmanTestsUtility.WaitForDeletedFile(cwd, "jquery.min.js", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void FileSaveRestore_Unpkg_DeleteLibrary()
+        {
+            _libManConfig.Open();
+            string[] expectedFilesAndFolders = new[] {
+                "LICENSE.txt",
+                "AUTHORS.txt",
+            };
+
+            Editor.Caret.MoveToExpression("\"version\": \"1.0\"");
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "unpkg", caseInsensitive: true);
+            Editor.KeyboardCommands.Type("unpkg");
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            _libManConfig.Save();
+            string cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib", "jquery");
+            LibmanTestsUtility.WaitForRestoredFiles(cwd, expectedFilesAndFolders, caseInsensitive: true);
+
+            Editor.Caret.MoveToExpression("{", 0, 2);
+
+            for (int i = 0; i < 4; ++i)
+            {
+                Editor.Edit.DeleteLine();
+            }
+
+            _libManConfig.Save();
+            LibmanTestsUtility.WaitForDeletedFiles(cwd, expectedFilesAndFolders, caseInsensitive: true);
+        }
```

Comment:
Shouldn't this just check that there are no files left under the path?  Instead of checking a hardcoded list?

Reply:
For that matter, we should be deleting the folder too, but I can't remember if that fix is submitted yet.

---
In reply to: [194555302](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194555302) [](ancestors = 194555302)

Reply:
Yes. It make sense. I can try verifying there's no files in it

---
In reply to: [194555373](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194555373) [](ancestors = 194555373,194555302)